crates/control/src/behavior/dribble.rs

```diff
@@ -102,15 +118,52 @@ pub fn execute(
     }
 }
 
-fn is_kick_pose_reached(
+pub fn is_kick_pose_reached(
     kick_pose: Pose2<Ground>,
-    kick_info: &InWalkKickInfoParameters,
+    thresholds: nalgebra::Vector3<f32>,
     ground_to_upcoming_support: Isometry2<Ground, UpcomingSupport>,
 ) -> bool {
     let upcoming_kick_pose = ground_to_upcoming_support * kick_pose;
-    let is_x_reached = upcoming_kick_pose.position().x().abs() < kick_info.reached_thresholds.x;
-    let is_y_reached = upcoming_kick_pose.position().y().abs() < kick_info.reached_thresholds.y;
-    let is_orientation_reached =
-        upcoming_kick_pose.orientation().angle().abs() < kick_info.reached_thresholds.z;
+
+    let is_x_reached = upcoming_kick_pose.position().x().abs() < thresholds.x;
+    let is_y_reached = upcoming_kick_pose.position().y().abs() < thresholds.y;
+    let is_orientation_reached = upcoming_kick_pose.orientation().angle().abs() < thresholds.z;
+
     is_x_reached && is_y_reached && is_orientation_reached
 }
+
+pub fn break_precision_kick(
+    game_controller_state: Option<FilteredGameControllerState>,
+    precision_kick_timeout: u8,
+    bigger_threshold_start_time: Option<SystemTime>,
+    cycle_start_time: SystemTime,
+) -> bool {
+    let game_controller_state = game_controller_state.unwrap_or_default();
+    let mut time_difference: Duration = Duration::default();
+
+    if bigger_threshold_start_time.is_some() {
```

Comment:
Just return false in the else case, then you also don't need the `mut time_difference`

Reply:
done